plugins/auth-backend/src/providers/microsoft/provider.test.ts

```diff
@@ -103,4 +128,45 @@ describe('createMicrosoftProvider', () => {
       },
     });
   });
+
+  it('should return the base64 encoded photo data of the profile', async () => {
+    setupHandlers();
+    const tokenIssuer = {
+      issueToken: jest.fn(),
+      listPublicKeys: jest.fn(),
+    };
+    const catalogIdentityClient = {
+      findUser: jest.fn(),
+    };
+
+    const provider = new MicrosoftAuthProvider({
+      logger: getVoidLogger(),
+      catalogIdentityClient:
+        catalogIdentityClient as unknown as CatalogIdentityClient,
+      tokenIssuer: tokenIssuer as unknown as TokenIssuer,
+      authHandler: async ({ fullProfile }) => ({
+        profile: {
+          email: fullProfile.emails![0]!.value,
+          displayName: fullProfile.displayName,
+          picture: 'http://microsoft.com/lols',
+        },
+      }),
+      clientId: 'mock',
+      clientSecret: 'mock',
+      callbackUrl: 'mock',
+      // define resolver to return user `info` for photo validation
+      signInResolver: async (info, _) => {
+        return {
+          id: 'user.name',
+          token: 'token',
+          info: info,
+        };
+      },
+    });
```

Comment:
@rugvip Added a mocked & intercepted image response!

To validate the photo on the identity, I specified a _signInResolve_ method that returns the user's info. This is a bit of a hack, but I wasn't sure how else to access the `fullProfile.photos` attribute. Lmk if there's something better. Thanks.